### PR TITLE
Remove site picker step from woocommerce-installation

### PR DESCRIPTION
**Pull request by lsl**
I can't find a way to make the site picker optional in https://github.com/Automattic/wp-calypso/pull/58536 so this PR is to remove it completely and we'll just need to rely on the site slug for all testing.

#### Changes proposed in this Pull Request

* Removes site picker step from woocommerce-installation

#### Testing instructions

Confirm transfer still works as expected

* http://calypso.localhost:3000/start/woocommerce-install/confirm?site=wooptestupgrade.wordpres\s.com

Related to #58229

### CODE REVIEW

File: client/signup/steps/woocommerce-install/index.tsx [hunk 1]
```diff
@@ -11,4 +11,7 @@ export interface WooCommerceInstallProps {
 	queryObject: {
 		siteSlug: string;
 	};
+	signupDependencies: {
+		siteSlug: string;
+	};
```
Comment (lsl) [inline client/signup/steps/woocommerce-install/index.tsx:16]: This isn't being used but wanted to give us some prior art for any other steps coming up in the next cycle with onboarding flows that will need to take advantage of how this works.

**Comment by retrofox** [comment]
Testing according to the instructions and it works as expected. I'm getting an issue though, that I'd like to confirm and ask whether we should try to handle it here, in this PR.
When I switch the testing site, there is a cache issue or something like that that ends up showing a different site ID that the site defined in the query string:

Screen Shot 2021-11-29 at 4 11 50 PM

Is it something expected?

**Comment by lsl** [comment]
> When I switch the testing site, there is a cache issue or something like that that ends up showing a different site ID that the site defined in the query string:

This is due to this [the ordering](https://github.com/Automattic/wp-calypso/blob/1ae846fd6d6e09cf7897a2e4aad5ce35bd14cb0e/client/signup/controller.js#L320-L324) we look at dependencies.

This feels like something we should get right in this PR.. it feels pretty hacky but I think if we set the selected site based on the query object value it should work.

File: client/signup/controller.js [hunk 2]
```diff
@@ -292,7 +292,8 @@ export default {
 		// If the flow has siteId or siteSlug as query dependencies, we should not clear selected site id
```
Comment (retrofox) [inline client/signup/controller.js:292]: probably we'd like to update this comment.

File: client/signup/steps/woocommerce-install/confirm/index.tsx [hunk 3]
```diff
@@ -141,7 +145,7 @@ export default function Confirm( props: WooCommerceInstallProps ): ReactElement
 			hideSkip={ true }
 			nextLabelText={ __( 'Confirm' ) }
 			allowBackFirstStep={ true }
-			backUrl="select-site"
+			backUrl={ `/woocommerce-installation/${ wpcomDomain }` }
```
Comment (retrofox) [inline client/signup/steps/woocommerce-install/confirm/index.tsx:148]: ~should we define the `confirm` step here? probably it's the same~

Comment (retrofox) [reply client/signup/steps/woocommerce-install/confirm/index.tsx:148]: nice

**Review by retrofox — APPROVED** [review]
LGTM. Tested and I can switch between different sites rightly.

**Pull request merged**